### PR TITLE
check fun

### PR DESCRIPTION
**Pull request by kevmoo**
- checks: String add length, is(Not)Empty, startsWith, endsWith
- checks: Add some TODOs around handling String in `literal` helper

### CODE REVIEW

**Comment by kevmoo** [comment]
Will land as seperate commits

File: pkgs/checks/lib/src/extensions/core.dart [hunk 1]
```diff
@@ -118,4 +118,34 @@ extension StringChecks on Check<String> {
...
+
+  void isNotEmpty() {
+    context.expect(() => const ['is not empty'], (actual) {
+      if (actual.isNotEmpty) return null;
+      return Rejection(actual: literal(actual), which: ['is empty']);
+    });
+  }
+
+  void startsWith(String other) {
+    context.expect(() => const ['starts with'], (actual) {
```
Comment (natebosch) [inline pkgs/checks/lib/src/extensions/core.dart:139]:
```suggestion
    context.expect(() => const ['starts with ${literal(other)}'], (actual) {
```

File: pkgs/checks/lib/src/extensions/core.dart [hunk 2]
```diff
@@ -118,4 +118,34 @@ extension StringChecks on Check<String> {
...
+      if (actual.startsWith(other)) return null;
+      return Rejection(actual: literal(actual), which: ['does not start with']);
```
Comment (natebosch) [inline pkgs/checks/lib/src/extensions/core.dart:141]:
```suggestion
      return Rejection(actual: literal(actual), which: ['does not start with ${literal(other)}']);
```

File: pkgs/checks/lib/src/extensions/core.dart [hunk 3]
```diff
@@ -118,4 +118,34 @@ extension StringChecks on Check<String> {
...
+    });
+  }
+
+  void endsWith(String other) {
+    context.expect(() => const ['ends with'], (actual) {
```
Comment (natebosch) [inline pkgs/checks/lib/src/extensions/core.dart:146]:
```suggestion
    context.expect(() => const ['ends with ${literal(other)}'], (actual) {
```

File: pkgs/checks/lib/src/extensions/core.dart [hunk 4]
```diff
@@ -118,4 +118,34 @@ extension StringChecks on Check<String> {
...
+      if (actual.endsWith(other)) return null;
+      return Rejection(actual: literal(actual), which: ['does not end with']);
```
Comment (natebosch) [inline pkgs/checks/lib/src/extensions/core.dart:148]:
```suggestion
      return Rejection(actual: literal(actual), which: ['does not end with ${literal(other)}']);
```

**Review by natebosch — APPROVED** [review]

**Pull request merged**